### ISSUE TITLE
TypeError when using `typing.Literal`

### ISSUE THREAD

**Issue by mar-muel** [pydantic/pydantic#6027]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))


### Description

Hi team! Thanks for your work, love this library.

I'm running into the following issue since very recently. The following code works locally

```python
# file test_config.py
from typing import Literal
from pydantic import BaseModel
import pydantic.utils

print(pydantic.utils.version_info())

class MyConfig(BaseModel):
    my_name: Literal['hello', 'world'] = 'hello'


cfg = MyConfig(my_name='hello')
print(cfg)
```

```
             pydantic version: 1.10.7
            pydantic compiled: True
                 install path: /Users/martin/miniconda3/envs/gal-worker/lib/python3.8/site-packages/pydantic
               python version: 3.8.16 (default, Jan 17 2023, 16:42:09)  [Clang 14.0.6 ]
                     platform: macOS-10.16-x86_64-i386-64bit
     optional deps. installed: ['dotenv', 'typing-extensions']
my_name='hello'
```

But then with Dockerfile and using the same Python version

Dockerfile:
```Dockerfile
FROM python:3.8.16
ENV PYTHONUNBUFFERED=1
RUN pip install pydantic==1.10.7
COPY test_config.py test_config.py
RUN python3 test_config.py
```

Then running `docker build .` gives me:
```
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                                                                                                                   0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                                                                                                                                                                                    0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                                                                                                                        0.0s
 => [internal] load metadata for docker.io/library/python:3.8.16                                                                                                                                                                                                                                                                                                       0.6s
 => [auth] library/python:pull token for registry-1.docker.io                                                                                                                                                                                                                                                                                                          0.0s
 => [1/4] FROM docker.io/library/python:3.8.16@sha256:e411647c253b75948394a343b13ff32b5674687df0c54187445d12ee9de2b106                                                                                                                                                                                                                                                 0.0s
 => [internal] load build context                                                                                                                                                                                                                                                                                                                                      0.0s
 => => transferring context: 305B                                                                                                                                                                                                                                                                                                                                      0.0s
 => CACHED [2/4] RUN pip install pydantic==1.10.7                                                                                                                                                                                                                                                                                                                      0.0s
 => [3/4] COPY test_config.py test_config.py                                                                                                                                                                                                                                                                                                                           0.0s
 => ERROR [4/4] RUN python3 test_config.py                                                                                                                                                                                                                                                                                                                             0.4s
------
 > [4/4] RUN python3 test_config.py:
#9 0.302              pydantic version: 1.10.7
#9 0.302             pydantic compiled: True
#9 0.302                  install path: /usr/local/lib/python3.8/site-packages/pydantic
#9 0.302                python version: 3.8.16 (default, May 23 2023, 14:26:40)  [GCC 10.2.1 20210110]
#9 0.302                      platform: Linux-5.10.47-linuxkit-x86_64-with-glibc2.2.5
#9 0.302      optional deps. installed: ['typing-extensions']
#9 0.303 Traceback (most recent call last):
#9 0.303   File "test_config.py", line 9, in <module>
#9 0.304     class MyConfig(BaseModel):
#9 0.304   File "pydantic/main.py", line 197, in pydantic.main.ModelMetaclass.__new__
#9 0.305   File "pydantic/fields.py", line 506, in pydantic.fields.ModelField.infer
#9 0.305   File "pydantic/fields.py", line 436, in pydantic.fields.ModelField.__init__
#9 0.306   File "pydantic/fields.py", line 552, in pydantic.fields.ModelField.prepare
#9 0.306   File "pydantic/fields.py", line 668, in pydantic.fields.ModelField._type_analysis
#9 0.306   File "/usr/local/lib/python3.8/typing.py", line 774, in __subclasscheck__
#9 0.307     return issubclass(cls, self.__origin__)
#9 0.308 TypeError: issubclass() arg 1 must be a class
------
executor failed running [/bin/sh -c python3 test_config.py]: exit code: 1
```

Given that this is a very recent issue, I wonder whether this is related to Python or some other upstream changes to `typing`?

### Example Code

_No response_

### Python, Pydantic & OS Version

```Text
pydantic version: 1.10.7
pydantic compiled: True
install path: /usr/local/lib/python3.8/site-packages/pydantic
python version: 3.8.16 (default, May 23 2023, 14:26:40)  [GCC 10.2.1 20210110]
platform: Linux-5.10.47-linuxkit-x86_64-with-glibc2.2.5
optional deps. installed: ['typing-extensions']
```


### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [X] [Data validation/parsing](https://docs.pydantic.dev/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/usage/schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/usage/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/usage/model_config/)
- [X] [Field Types](https://docs.pydantic.dev/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/usage/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by mar-muel** [comment]
I could now also reproduce it locally by creating a new python env with conda:

```bash
conda create -n testenv python=3.8.16
conda activate testenv
pip install pydantic==1.10.7

python test_config.py
```

gives
```
             pydantic version: 1.10.7
            pydantic compiled: True
                 install path: /Users/martin/miniconda3/envs/testenv/lib/python3.8/site-packages/pydantic
               python version: 3.8.16 (default, Mar  1 2023, 21:19:10)  [Clang 14.0.6 ]
                     platform: macOS-10.16-x86_64-i386-64bit
     optional deps. installed: ['typing-extensions']
Traceback (most recent call last):
  File "test_config.py", line 7, in <module>
    class MyConfig(BaseModel):
  File "pydantic/main.py", line 197, in pydantic.main.ModelMetaclass.__new__
  File "pydantic/fields.py", line 506, in pydantic.fields.ModelField.infer
  File "pydantic/fields.py", line 436, in pydantic.fields.ModelField.__init__
  File "pydantic/fields.py", line 552, in pydantic.fields.ModelField.prepare
  File "pydantic/fields.py", line 668, in pydantic.fields.ModelField._type_analysis
  File "/Users/martin/miniconda3/envs/testenv/lib/python3.8/typing.py", line 774, in __subclasscheck__
    return issubclass(cls, self.__origin__)
TypeError: issubclass() arg 1 must be a class
```

It seems like the actual error is when running the statement 🤔 

```python
issubclass(typing.Tuple, tuple)
```

**Comment by dmontagu** [comment]
This has been fixed in pydantic 1.10.8 — it's caused by newer versions of the `typing_extensions` module.

If upgrading to `pydantic==1.10.8` doesn't fix the issue for you, please let me know and I'll reopen this issue.
